cmd/deploy.go

```diff
@@ -254,19 +254,29 @@ func runDeploy(cmd *cobra.Command, newClient ClientFactory) (err error) {
 	if !f.Initialized() {
 		if !cfg.Remote || f.Build.Git.URL == "" {
 			// Only error if this is not a fully remote build
-			return fmt.Errorf(`no function found in current directory.
-You need to be inside a function directory to deploy it.
+			// Layer 2: Wrap technical error with CLI-specific guidance
+			var errNotInit *fn.ErrNotInitialized
+			notInitErr := fn.NewErrNotInitialized(f.Root)
+			if errors.As(notInitErr, &errNotInit) {
+				return fmt.Errorf(`%w
+
+To deploy a function, you need to point to a function directory or use a --path flag approach.
```

Comment:
I think this was intended to be 
```suggestion
To deploy a function, you need to be in a function directory or use a --path flag approach.
```